File: oop-backend/src/main/java/oop_backend/oop/model/Player.java
package oop_backend.oop.model;

public class Player {
    private Long userId;
    private String countryName;
    private int rating;
    private String policy;
    private int economyScore; // Yeni ekonomi skoru alanı
    private int welfareScore; // Yeni refah skoru alanı
    
    // Yapıcılar
    public Player() {
        this.rating = 0; // Başlangıç değeri
        this.economyScore = 31; // Başlangıç değeri
        this.welfareScore = 40; // Başlangıç değeri
    }
    
    public Player(Long userId, String countryName) {
        this.userId = userId;
        this.countryName = countryName;
        this.rating = 0; // Başlangıç değeri
        this.economyScore = 31; // Başlangıç değeri
        this.welfareScore = 50; // Başlangıç değeri
    }
    
    // Getter ve setter metodları
    public Long getUserId() {
        return userId;
    }
    
    public void setUserId(Long userId) {
        this.userId = userId;
    }
    
    public String getCountryName() {
        return countryName;
    }
    
    public void setCountryName(String countryName) {
        this.countryName = countryName;
    }
    
    public int getRating() {
        return rating;
    }
    
    public void setRating(int rating) {
        this.rating = rating;
    }
    
    public String getPolicy() {
        return policy;
    }
    
    public void setPolicy(String policy) {
        this.policy = policy;
    }
    
    public int getEconomyScore() {
        return economyScore;
    }
    
    public void setEconomyScore(int economyScore) {
        // Sınırları kontrol et (0-100 arası)
        this.economyScore = Math.max(0, Math.min(100, economyScore));
    }
    
    public int getWelfareScore() {
        return welfareScore;
    }
    
    public void setWelfareScore(int welfareScore) {
        // Sınırları kontrol et (0-100 arası)
        this.welfareScore = Math.max(0, Math.min(100, welfareScore));
    }
}

File: oop-frontend/src/models/Player.js
export class Player {
    constructor(data = {}) {
        this.userId = data.userId || null;
        this.countryName = data.countryName || '';
        this.rating = data.rating || 0;
        this.policy = data.policy || '';
    }

   
    getUserId() {
        return this.userId;
    }

    setUserId(userId) {
        this.userId = userId;
    }

    getCountryName() {
        return this.countryName;
    }

    setCountryName(countryName) {
        this.countryName = countryName;
    }

    getRating() {
        return this.rating;
    }

    setRating(rating) {
        this.rating = rating;
    }

    getPolicy() {
        return this.policy;
    }

    setPolicy(policy) {
        this.policy = policy;
    }

    static fromApi(apiData) {
        return new Player(apiData);
    }

    static listFromApi(playerDataArray) {
        if (!playerDataArray || !Array.isArray(playerDataArray)) {
            console.error('Expected player array but received:', playerDataArray);
            return []; 
        }
        
        return playerDataArray.map(playerData => Player.fromApi(playerData));
    }

    
    toApiFormat() {
        return {
            userId: this.userId,
            countryName: this.countryName,
            rating: this.rating
        };
    }
}